### PR TITLE
 Remove unused packages and tests

### LINKED ISSUE #569: Add an option to keep non-go files to `dep`
# Problem

Currently `dep prune` removes non-go files from all the packages. That doesn't work for us because some packages we use have C sources in it. In the next release of `dep` there [would be an option](https://github.com/golang/dep/pull/1226) to keep non-go files intact.

When it is released, let's use it and use `dep prune` to keep our vendor folder nice and slim.

# Acceptance Criteria

1. Essential non-go files aren't touched when `dep prune` is called.
2. `Gopkg.toml` has `[prune]` section with options set.
3. Updated (after `dep prune`) `vendor` folder is committed. Tests are removed, unused packages are removed.
4. Tests pass and the library still builds.

### PR DESCRIPTION
**Pull request by dshulyak**
Fixes: https://github.com/status-im/status-go/issues/569
Follow up for: https://github.com/status-im/status-go/pull/551

Dep with required fix was released: https://golang.github.io/dep/blog/2018/01/23/announce-v0.4.0.html

### CODE REVIEW

File: DEPENDENCIES.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Dependency Management
+[`dep`](https://github.com/golang/dep) is a tool of choice when it comes to dependency management.
+
+## How we use `dep`.
+
+1. Transitive dependencies of `go-ethereum`. The most important thing for us is
+   to be in-sync there. We want to reduce the regression scope. 
+   Hence, we pin down all the dependencies of `go-ethereum` with SHAs in `Gopkg.toml` when
+   importing a new version of upstream. (This is considered a bad practice for
+   `dep` but we are willing
```
Comment (mandrigin) [inline DEPENDENCIES.md:10]: It is based on an outdated version of #551. I think it is worth rebasing.

**Review by mandrigin — CHANGES_REQUESTED** [review]
1 note, looks good otherwise 👍

File: DEPENDENCIES.md (hunk 1 above)
Comment (dshulyak) [reply DEPENDENCIES.md:10]: I will cherry pick on develop when your change will be merged

**Comment by mandrigin** [comment]
@dshulyak should we really keep 2 PRs then? ~Let's merge this one and close #551?~

File: DEPENDENCIES.md (hunk 1 above)
Comment (mandrigin) [reply DEPENDENCIES.md:10]: ah, ok. but maybe I should cherry-pick this one to my branch instead? so we don't add and then remove tons of files?

Comment (dshulyak) [reply DEPENDENCIES.md:10]: good idea, i will close this PR

**Pull request closed without merging**